add beginning structure for iterative example (#37)
fhdsl/DataViz_Considerations: 07_wrapping_up.qmd

ottrpal::set_knitr_image_path()
```

## Learning Objectives

```{r los, fig.align = "center", echo = FALSE, out.width="100%", fig.alt = "This chapter will demonstrate how to iteratively construct a data visualization, evaluate whether a data visualization can be improved to enhance its clarity, accessibility, or accuracy, and to discuss ways to hone data visualization skills"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fu-KfdN2ldOXB49o9zdpurFbVvl1bOh8cfQ3vu0szk4/edit?slide=id.g3bc553a32e9_0_0#slide=id.g3bc553a32e9_0_0")
```

## Iterative Example

To summarize all of the considerations and best practices discussed in the chapters before this, the first section of this final chapter will walk through an example, step-by-step, of iteratively building up an expository data visualization.

### Data Introduction

::: {.panel-tabset}

#### Description

A [2017 survey from the University of British Columbia](https://www.scq.ubc.ca/wp-content/uploads/2017/10/candyhierarchysurvey2017.pdf) asked potential trick or treaters a series of questions. The main question provided a list of candy (and a few joke non-candy items) to survey takers. The prompt asked

> >Basically, consider that feeling you get when you receive this item in your Halloween haul. Does it make you really happy (joy)? Or is it something that you automatically

Reviewer: ```suggestion
>Basically, consider that feeling you get when you receive this item in your Halloween haul. Does it make you really happy (joy)? Or is it something that you automatically place in the junk pile (despair)? Meh for indifference, and you can leave blank if you have no idea what the item is.
```

Reviewer: putting this on one line so it keeps the quote formatting

> place in the junk pile (despair)? Meh for indifference, and you can leave blank if you have no idea what the item is.

Reviewer: ```suggestion
```

Reviewer: putting this with the previous line in the next comment

> Using this [dataset](https://www.scq.ubc.ca/so-much-candy-data-seriously/), we focused on responses to the question about feelings related to specific candies, selecting only real candy, for a total of 76 candies. For each of these candies, we computed the proportion of respondents who replied with each general feeling (e.g., Joy, Despair, Indifference, or No response).

Reviewer: maybe switch to present tense language here... we will focus on specific candies...etc. and and more about what we are about to do overall

```suggestion
Using this [dataset](https://www.scq.ubc.ca/so-much-candy-data-seriously/), we will focus on responses to the question about feelings related to specific candies, selecting only real candy, for a total of 76 candies. For each of these candies, we will compute the proportion of respondents who replied with each general feeling (e.g., Joy, Despair, Indifference, or No response). We will use this to make a visualization about how respondents feel about different kinds of candy.
```

#### R code

> ```{r}

Reviewer: the code was already expanded in the preview but perhaps that is just something about the preview

Author: @carriewright11 That's the current setting I'm using to have the code snippets expanded by default. `code-fold: show`. Would it be better for these to be collapsed by default?

Reviewer: I think so - we want people to focus on your overall message not the code in this particular case

Reviewer: But also I only think it matters much for this first one actually I like it expanded for the others… if we expand more later about the variables then maybe it’s ok as is

Author: The first one of importing the data and wrangling it is pretty long. I can collapse those for sure

#| label: importR
#| warning: false
#| message: false
#| code-fold: show
#| code-summary: "R code for data import and preliminary wrangling steps"

library(here)
library(tidyverse)

df_2017 <- read.csv(here("data/candyhierarchy2017.csv"), fileEncoding = "ISO-8859-1")

> non_candy <- c("Bonkers  the board game",

Reviewer: the spacing looks a tad funny on these, but I imagine that was original to the dataset?

Author: Yes -- the reading in of the dataset with R does some weird spacing and special characters for a few columns

               "Box o Raisins",
               "Broken glow stick",
               "Cash  or other forms of legal tender",
               "Chardonnay",
               "Creepy Religious comics Chick Tracts",
               "Dental paraphenalia",
               "Generic Brand Acetaminophen",
               "Glow sticks",
               "Healthy Fruit",
               "Hugs  actual physical hugs",
               "JoyJoy  Mit Iodine",
               "Kale smoothie",
               "Senior Mints",
               "Green Party M M s",
               "Independent M M s",
               "Abstained from M M ing",
               "Minibags of chips",
               "Pencils",
               "Real Housewives of Orange County Season 9 Blue Ray",
               "Sandwich sized bags filled with BooBerry Crunch",
               "Spotted Dick",
               "Trail Mix",
               "Vials of pure high fructose corn syrup  for main lining into your vein",
               "Vicodin",
               "White Bread",
               "Whole Wheat anything")

> #making a dataframe with just candy feeling responses

Reviewer: ```suggestion
## making a dataframe of just the feeling responses for the candy
```

Reviewer: just making this match and making it slightly more clear

## polishing the column names to remove "Q6", "." and white space
## selecting columns that are just the candy by excluding all of the non-candy
## using pivot_longer to make a column that has candy names (`column_name`) and responses of "JOY", "DESPAIR", or "MEH" in `value`
## each candy will have multiple rows because of pivot_longer use
## grouping by candy and the value to count (using `summarise`) the number of respondents who felt Joy, Despair, or Meh for each candy
## dropping the value grouping to sum counts to find total counts for each candy
## undoing all of the grouping
## finding a proportion for each row: candy and feeling combo

df_2017_jc <- df_2017 %>%
  select(starts_with("Q6")) %>%
  rename_with(~ str_remove(., "Q6...")) %>%
  rename_with(~ str_replace_all(., "\\.", " ")) %>%
  rename_with(~ str_trim(.x, side = "both")) %>%
  select(-all_of(non_candy)) %>%
  pivot_longer(everything(), names_to = "column_name", values_to = "value") %>%
  group_by(column_name, value) %>%
  summarise(count = n(), .groups = "drop_last") %>%
  mutate(total = sum(count)) %>%
  ungroup() %>%
  mutate(proportion = count / total)

> # replacing empty values with NA

Author: ```suggestion
## replacing empty values with NA
```

Author: parallelizing like other comments

                                                    "Independent M&M's",
                                                    "Abstained from M&M'ing.",
                                                    "Minibags of chips",
                                                    "Pencils",
                                                    "Real Housewives of Orange County Season 9 Blue-Ray",
                                                    "Sandwich-sized bags filled with BooBerry Crunch",
                                                    "Spotted Dick",
                                                    "Trail Mix",
                                                    "Vials of pure high fructose corn syrup, for main-lining into your vein",
                                                    "Vicodin",
                                                    "White Bread",
                                                    "Whole Wheat anything"])]

just_candy_props.shape[0]

```

:::

:::{.emphasis_block}

<div class = "reflection">
What variables (specifically what types of variables -- categorical or numerical) are part of this dataset as described?
</div>

> <details><summary>Variables in the candy dataset</summary>

Reviewer: I actually think it would be better if this was just expanded and not in an expandable section

Reviewer: but I also get why you had it this way- just not sure everyone will click it and might be disoriented - maybe could explain what the variables are but not the type? and put that in an expansion?

Author: @carriewright11 That was a question I generally had about the reflection boxes. Definitely not sure if it's better to just have the answers after the question without the need to click to expand, or better to keep the click to expand, or some hybrid like you're suggesting (which I like the idea of). Is there a way you've done it with other courses?

Reviewer: I think it depends, I think we just might need some more orienting about the variables for folks who don’t engage with the questions if that makes sense

Reviewer: Sometimes activities are more obvious - in this case it’s a new concept for the course so we have to expect some people not to unlike a workshop

Author: That definitely makes sense. I'll workshop this a bit then so there's some more orienting about the variables

ottrpal::include_slide("https://docs.google.com/presentation/d/1fu-KfdN2ldOXB49o9zdpurFbVvl1bOh8cfQ3vu0szk4/edit?slide=id.g3bb275f27df_0_6#slide=id.g3bb275f27df_0_6")
```

### Choosing a plot type

::: {.emphasis_block}

<div class="reflection">
Given the data that we have to work with and the goal of the visualization, what type of plot would you use?
</div>

<details><summary>Comparing and contrasting plot types for this visualization</summary>

Note that respondents did not simply rank the candy based on preference and we instead have two numerical values for each candy. Therefore we should probably consider plots types beyond the conventional ranking plots described in Chapter 4.

**A more conventional approach**:

A bar plot or even a stacked bar plot could work if we computed a single score (proportion of Joy - proportion of Despair perhaps), but with 76 candies, we would need to arrange the bars based on the magnitude or size and just focus on the bars on the extremes. The visualization would likely appear pretty cramped overall if we tried to show every candy.

>   * Here's an example of an analysis that utilized bar plots: [https://github.com/phoebewong/candy-hierarchy-2017](https://github.com/phoebewong/candy-hierarchy-2017). Notice, for this example they focused on a couple specific candies which gets around the "cramped" visualization concern.

Reviewer: love this!

  * Similar to a bar plot is a lollipop chart (seems fitting for a candy dataset): [https://x.com/ttrodrigz/status/923582440937021440](https://x.com/ttrodrigz/status/923582440937021440). This example uses a log ratio rather than a difference for each candy, sorts based on magnitude, and then focuses on the extremes as expected.

**A more unconventional approach**:

A scatter plot is constructed with two numerical variables and is often used to show a correlation. We might expect our data to have some sort of a correlation because likely candy that is greatly enjoyed by many won't be despised by many. And conversely, candy that is despised by many probably won't be greatly enjoyed. The only way to see if this expectation is true is to explore the data and just plot it. So that's what we'll do for the first iteration.

</details>

:::

<div class="notice">
To review different conventional plot types, visit [Chapter 4](https://hutchdatascience.org/DataViz_Considerations/04_graph_info.html#standard-graph-types).
</div>

### Building the plot

> The first few iterations will be an exploratory analysis of the data where we don't focus on aesthetics, but rather

Reviewer: ```suggestion
The first few iterations will be an exploratory analysis of the data where we don't focus on aesthetics, but rather:
```

3. Using list comprehension to build an input to the labeling function of what the labels are and where they're going to go. Only labeling one out of every three candies to avoid overcrowding.
4. Using the adjust_text function from the adjustText package to automatically adjust the position of text labels to minimize overlaps

<!--
The image embedded below was saved locally after running the above code manually. Note that it does NOT autoupdate if any changes are made to the content.
-->

![](resources/images/candy_iter2_python.png){fig-alt="Python plot that labels the candy names"}

:::

::: {.emphasis_block}

<div class="reflection">
What do you notice after observing the labels? Do you have any ideas on new variables or a takeaway message?
</div>

<details><summary>Takeaways from the candy name labeling</summary>
It looks like chocolate candy (Twix, Kit Kat, 100 Grand Bar...) is among the most liked candy. It could be beneficial to create a new categorical variable that represents if each candy is chocolate or not.
</details>
:::

#### Iteration 3

::: {.panel-tabset}

##### Description

> ```{r iter3Plot, fig.align = "center", echo = FALSE, out.width="100%", fig.alt = "This iteration adds color to distinguish groups, specifically chocolate versus not chocolate groups. Light blue represents chocolate while light pink represents not chocolate candy. We see a fairly clear separation with chocolate on one extreme and non-chocolate on the other with mixing of the groups in the middle."}

Reviewer: ```suggestion
```{r iter3Plot, fig.align = "center", echo = FALSE, out.width="100%", fig.alt = "This iteration adds color to distinguish groups, specifically chocolate versus not chocolate groups. Light blue represents chocolate, while light pink represents candy that is not chocolate. We see a fairly clear separation with chocolate on one extreme and non-chocolate on the other with mixing of the groups in the middle."}
```

```

#### Iteration 20

```{r iter20Pot, fig.align = "center", echo = FALSE, out.width="100%"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fu-KfdN2ldOXB49o9zdpurFbVvl1bOh8cfQ3vu0szk4/edit?slide=id.g3be338b6b90_0_99#slide=id.g3be338b6b90_0_99")
```

```{r iter20bPlot, fig.align = "center", echo = FALSE, out.width="100%"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fu-KfdN2ldOXB49o9zdpurFbVvl1bOh8cfQ3vu0szk4/edit?slide=id.g3c26f54f9da_0_8#slide=id.g3c26f54f9da_0_8")
```

#### The plot!

```{r finalPlot, fig.align = "center", echo = FALSE, out.width="100%"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fu-KfdN2ldOXB49o9zdpurFbVvl1bOh8cfQ3vu0szk4/edit?slide=id.g3be338b6b90_0_146#slide=id.g3be338b6b90_0_146")
```

But we're not done yet....

#### The caption!

```{r plotCaption, fig.align = "center", echo = FALSE, out.width="100%"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fu-KfdN2ldOXB49o9zdpurFbVvl1bOh8cfQ3vu0szk4/edit?slide=id.g3bb275f27df_0_20#slide=id.g3bb275f27df_0_20")
```

#### Going further

> The dataset is an open dataset meaning that it is available for you to explore and visualize on your own!

Reviewer: ```suggestion
The dataset is an open dataset, meaning that it is available for you to explore and visualize on your own!
```
